### PR TITLE
Add JSON schema spell and sensitivity checkers

### PR DESCRIPTION
**Pull request by cbetta**
I am not sure if these are in the right category. I'd love some recommendations on this.

### CODE REVIEW

File: _data/tools.yml [hunk 1]
```diff
@@ -1034,6 +1034,24 @@
   description: Generate OpenAPI 3 specification for Laravel Applications.
   v2: false
   v3: true
+  
+- name: json-schema-spell-checker
+  category:
+    - miscellaneous
+  language: Node.js
+  github: https://github.com/mheap/json-schema-spell-checker
+  description: Check OpenAPI specification for spelling mistakes using mdspell
```
Comment (philsturgeon) [inline _data/tools.yml]:
```suggestion
  description: Check OpenAPI descriptions for spelling mistakes using mdspell.
```

**Comment by philsturgeon** [comment]
Thanks @cbetta. I have to ask first, is it OpenAPI or JSON Schema? They're not the same thing. OpenAPI has a Schema Object _"based"_ on JSON Schema, but [they are not interchangeable](https://apisyouwonthate.com/blog/openapi-and-json-schema-divergence-part-1). Not yet anyway (https://github.com/OAI/OpenAPI-Specification/pull/1977). 

Same question to @mheap.

**Comment by cbetta** [comment]
Ah, good point. I think the answer is "both and more". This tool isn't really specific to either and instead can be used to validate any fields in an object that can be addressed using a JSON Path.

File: _data/tools.yml (hunk 1 above)
Comment (cbetta) [reply _data/tools.yml]: It can technically check anything. I use it to check titles and descriptions.

Comment (philsturgeon) [reply _data/tools.yml]: Then maybe it should just be called `json-spell-checker` instead? Then another tool called openapi-spell-checker could be implementing json spell checker on known openapi keywords, like title, description, summary, etc for various components. What do you think?

Comment (philsturgeon) [reply _data/tools.yml]: Oh I dont mean description fields, i mean API Descriptions. Check it out https://github.com/openapi-contrib/glossary

**Comment by philsturgeon** [comment]
This tool might be better named json-spell-checker to avoid confusion then? It could be cool to then leverage that tool with a thin layer which just points it at known OpenAPI fields like summary, description, etc and maybe even try spell checking enums as an option, etc. I'd rather not have generic JSON tooling on here even if it is useful for OpenAPI people, otherwise there's a _lot_ of stuff which could appear on here.

**Comment by cbetta** [comment]
Sure.

As for the name: I based the name on @mheap's original project. Maybe we should change the name if we know it will get merged in with a new name.

As for listing generic JSON tools: I see the point yet I would like to point out that @matthewtrask requested these tools to be added to this list 😆.

https://twitter.com/matthewtrask/status/1205135699256762369

Let me know if you think it's worth listing these libraries if we'd change the name or not.

**Comment by philsturgeon** [comment]
Yes, if then name was json-spell-checker I'd merge it. 

Bonus points if you make openapi-spell-checker which implements spell checking in all the places OpenAPI needs it (and can even handle markdown in places where OpenAPI handles markdown.

I'll take either but I think the combo is the most useful! :D

**Comment by daveshanley** [comment]
This project looks abandoned.

**Comment by mheap** [comment]
Abandoned, or stable? 😁

**Comment by daveshanley** [comment]
This looks abandoned to me. 

Screenshot 2023-06-04 at 8 56 21 AM
Screenshot 2023-06-04 at 8 56 16 AM

**Comment by philsturgeon** [comment]
@mheap as a comprimise to get this old old PR over the line, could you merge a few of those dependabot PRs and tag a release, then I'll be happy to merge this PR as is.

**Pull request closed without merging**